code/modules/surgery/advanced/toxichealing.dm

```diff
@@ -24,6 +24,12 @@
 /datum/surgery_step/toxichealing/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
 	user.visible_message("[user] starts rejuvenating some of [target]'s flesh back to life.", "<span class='notice'>You start knitting some of [target]'s flesh back to life.</span>")
 
+/datum/surgery_step/toxichealing/initiate(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, try_to_fail = FALSE)
+	if(..())
+		while((target.getToxLoss() >= 1) || (target.getOxyLoss() >= 1))
+			if(!..())
```

Comment:
```suggestion
			. = ..()
			if(!.)
```

Reply:
probably better this way, since then it'll return the correct value

Reply:
I just took it off the heal surgery, is that one wrong too?